packages/react-components/react-datepicker-compat/src/components/CalendarYear/CalendarYear.tsx

```diff
@@ -377,26 +377,26 @@ function useAnimateBackwards({ selectedYear, navigatedYear }: CalendarYearProps)
   }
 }
 
-const NavigationDirection = {
-  Previous: 0 as const,
-  Next: 1 as const,
-};
-
 function useYearRangeState({ selectedYear, navigatedYear }: CalendarYearProps) {
-  const [fromYear, navigate] = React.useReducer(
-    (state: number, action: (typeof NavigationDirection)[keyof typeof NavigationDirection]): number => {
-      return state + (action === NavigationDirection.Next ? CELL_COUNT : -CELL_COUNT);
-    },
-    undefined,
-    () => {
-      const rangeYear = selectedYear || navigatedYear || new Date().getFullYear();
-      return Math.floor(rangeYear / 10) * 10;
-    },
-  );
-  const toYear = fromYear + CELL_COUNT - 1;
+  const rangeYear = React.useMemo(() => {
+    return selectedYear || navigatedYear || Math.floor(new Date().getFullYear() / 10) * 10;
+  }, [navigatedYear, selectedYear]);
 
-  const onNavNext = () => navigate(NavigationDirection.Next);
-  const onNavPrevious = () => navigate(NavigationDirection.Previous);
+  const [fromYear, setFromYear] = React.useState<number>(rangeYear);
+
+  const onNavNext = () => {
+    setFromYear(i => i + CELL_COUNT);
+  };
+
+  const onNavPrevious = () => {
+    setFromYear(i => i - CELL_COUNT);
+  };
```

Comment:
_nit:_ Use relevant variable names

```suggestion
  const onNavNext = () => {
    setFromYear(year => year + CELL_COUNT);
  };

  const onNavPrevious = () => {
    setFromYear(year => year - CELL_COUNT);
  };
```